### DISCUSSION TITLE
Can't find ribbon button. How to change the order from ascending to descending by-metadata

### DISCUSSION THREAD

**Discussion by SebastianMC** [SebastianMC/obsidian-custom-sort#21] [Q&A] [answered]
Two questions were posted and addressed here:
- Can't find ribbon button
- How to change the sorting order direction from ascending to descending, for by-metadata order

**Comment by yyy3332** [comment]
cant find - ribbon button

**Reply by yyy3332** [reply]
<img width="1438" alt="Снимок экрана 2023-02-24 в 21 42 44" src="https://user-images.githubusercontent.com/47197223/221298830-05dfc37e-5b7e-449b-a034-23d0374709f0.png">

**Reply by SebastianMC** [reply]
Hi @yyy3332 

there are two common cases for the ribbon button not appearing:
1. Plugin is installed and not enabled after installation.
    - to fix this, go to the _Community plugins_ section of _Settings_ and use the respective checkbox to enable the plugin: \
![enable plugin](https://user-images.githubusercontent.com/23032356/221892895-01a82cf1-c3e1-467e-b9f5-d9ec809764d8.jpg)
2. Some generic failure on the plugin startup. Rare case, yet possible
    - to fix this a screenshot of Obsidian console should be posted here in order to help to identify the problem.

**Reply by yyy3332** [reply]
@SebastianMC 
Everything working now, thanks a lot man 💪❤️

**Reply by SebastianMC** [reply]
Good to hear, cool!

**Comment by Bojji-git** [comment]
Hello @SebastianMC, 

I recently installed your Custom File Explorer Sorting plugin. Somehow, I managed to get it do what I needed it to do (at last—sigh!). 

Now, here's where I'm stuck: I'm using the "< a-z by-metadata" type of sorting—it sorts notes having given Metadata alphabetically and in ascending order. That said, what I want is the exact opposite—I want to have them sorted alphabetically but in descending order.

So... in short, I want to know what the right wording equivalent to "order-desc: a-z" that should go where "a-z" is in "< a-z by-metadata" is, in order that my notes w given metadata be presented in alphabetical and descending order. I went through the manuals but couldn't find the wording. Could you help me with that, please?

Ps. Had to create a git acc. just so I could pose this message :).

**Reply by SebastianMC** [reply]
Hi @Bojji-git 

for clarity, the syntax which you currently use for alphabetical ascending sorting by metadata is (based on your example):
`< a-z by-metadata: my-metadata-field-name`

In the above syntax:
- the token `<` means _ascending order_
- the token `a-z` expresses _alphabetical_
- the token `by-metadata:` tells to sort by metadata value (of the metadata specified by name)

You want to change the direction from _ascending_ to _descending_ and to do that you need to replace the first token.
There are two valid syntaxes for the descending order:
- `>` which is the concise one, or
- `order-desc:` which is the verbose, more readable equivalent

In result, your desired syntax for alphabetical sorting by metadata with the order set to descending (instead of ascending) is:
`> a-z by-metadata: my-metadata-field-name`, or with the more verbose notation:
`order-desc: a-z by-metadata: my-metadata-field-name`

I hope it helps!

PS. Thanks for the report and the effort to register in github! I believe such Q & A exchanges (like this one) could be valuable and helpful for other people seeking advice. Longer term, the documentation will contain more HOW-TOs and real-life examples. Until then, my focus is to prepare next release of the plugin, with the support for order items via drag & drop in Bookmarks core plugin.

**Reply by Bojji-git** [reply, marked as answer]
Oh... That simple! That's the exact response I was looking for. Helped tremendously!

Thank you for your very clear explanation and for spearheading such a much needed plugin.

Looking forward to the new releases (though already satisfied with what's possible to accomplish thus far).
